packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -372,3 +419,53 @@ class LibraryCollection {
   bool isOtherLibrary(String classId) =>
       findOtherLibrary(displayClasses[classId]) != null;
 }
+
+class MemoryTimeline {
+  final List<HeapSample> data = [];
+
+  /// Notifies that a new Heap sample has been added to the timeline.
+  ValueListenable<HeapSample> get sampleAddedNotifier => _sampleAddedNotifier;
```

Comment:
This can just be
```dart
final ValueListenable<bool> = ValueNotifier<bool>(false);
```

Similar below.

Reply:
I actually think it is better how it currently is, so that consumers of `sampleAddedNotifier` cannot modify value - only read it. And then within this class `_sampleAddedNotifier.value` can be accessed to set the value. This is how it is done for listeners in the Timeline and Perf pages.

Reply:
Done.

Reply:
That's a good point, Kenzie.  Sounds good to me.

Just be sure to include the type declaration on the public exposed API.